feat: OPTIC-1970: Improve the usage of organization membership api requests on the client

bmartel opened the pull request:
This pull request introduces significant improvements to the query caching mechanism by centralizing query client configuration, enhancing cache management, and integrating TanStack Query for better consistency and maintainability. The changes also include updates to the API proxy to leverage the new caching approach.

### Query Client Centralization and Configuration
* Replaced the local `LSQueryClient` with a shared `queryClient` imported from `@humansignal/core/lib/utils/query-client`, which includes configurable cache time and a bypass mechanism for e2e testing. (`web/apps/labelstudio/src/app/App.jsx`, [[1]](diffhunk://#diff-36c1030ca59fcd90ceb45a672287abc0f8a01670167dc4d269eb7f44425b85b0L20-L38); `web/apps/labelstudio/src/utils/query-client.ts`, [[2]](diffhunk://#diff-ee25bb24a0248d145387d5135d7dd4843918ec9d1d27f8915da7c0d2c9f85f7bL1-R1); `web/libs/core/src/lib/utils/query-client.ts`, [[3]](diffhunk://#diff-7bcaae23ad881cad9748a6fa78ebd3852b843e9cef4e5b034a2712c699c37dbfR1-R21)

### API Proxy Enhancements
* Updated the `APIProxy` class to use TanStack Query's caching capabilities for GET requests, replacing the previous custom cache implementation. This includes using `queryClient.fetchQuery` with support for cache keys, stale times, and bypassing the cache when necessary. (`web/libs/datamanager/src/utils/api-proxy/index.js`, [[1]](diffhunk://#diff-3766ec5d32934dbe83c7088473bbe29602325c9082406621d1574d6b07e656e5L158-R184) [[2]](diffhunk://#diff-3766ec5d32934dbe83c7088473bbe29602325c9082406621d1574d6b07e656e5L290-R304)
* Removed the `shouldUseQueryCache` flag and replaced it with a more flexible `queryCacheParams` object to manage cache behavior. (`web/libs/datamanager/src/utils/api-proxy/index.js`, [[1]](diffhunk://#diff-3766ec5d32934dbe83c7088473bbe29602325c9082406621d1574d6b07e656e5L158-R184) [[2]](diffhunk://#diff-3766ec5d32934dbe83c7088473bbe29602325c9082406621d1574d6b07e656e5L265-L268)

### Cache Usage in Data Fetching
* Updated the `fetchUsers` method in `AppStore` to use the new caching system, specifying a cache key prefix and stale time for organization member data. (`web/libs/datamanager/src/stores/AppStore.js`, [web/libs/datamanager/src/stores/AppStore.jsL550-R555](diffhunk://#diff-d1e11a839827428aa6aada8e027a22b057a463ccb73c693e464c6e0784a9825fL550-R555))

These changes simplify the codebase, improve maintainability, and enhance the flexibility of caching mechanisms across the application.

bmartel committed:
feat: OPTIC-1970: Improve the usage of organization membership api requests on the client

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1970

yyassi-heartex commented on web/libs/datamanager/src/utils/api-proxy/index.js:
```diff
@@ -155,29 +156,32 @@ export class APIProxy {
-          delete finalParams.__useQueryCache;
+        if (__useQueryCache) {
+          queryCacheParams = {
+            staleTime: __useQueryCache.staleTime ?? undefined,
+            keyPrefix: __useQueryCache.prefixKey ?? methodSettings.path,
```
is there a chance `__useQueryCache` or `methodSettings` would be undefined? that could cause an expected error

bmartel replied:
`methodSettings` no, as it is defined beforehand manually, `__useQueryCache` yes, which is why the usage of this cache depends on there being a defined `__userQueryCache` property. If it is falsey, it will just retain the existing behaviour of the call as before.

bmartel committed:
refactor the FF which was surrounding the original change and use it here

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1970

bmartel committed:
rebuild lockfiles

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14856942441)

AndrejOros committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14856942441

bmartel commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14864575023)

bmartel committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14864575023

bmartel committed:
Merge branch 'develop' into 'fb-optic-1970'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14864575023

bmartel commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14864859167)

bmartel committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14864859167
